docs/Testing-GitHub-Actions.md

```diff
@@ -0,0 +1,142 @@
+
+#### During Action Authoring:
+
+- Running Unit test using VS Code / VS IDE ( Details to be added )
+- Running functional test from dev box ( Details to be added )
+
+#### Pull Request
+
+During pull request, we want to ensure action logic is tested using unit tests on all platform matrices and there is required code coverage. We have designed the YAML to ensure these coverages. Templates for actions will be set up to run the unit tests for the master and release branches.
+
+- Yaml file for PR is configured to do the following:
+  - Run on pull requests in master and release branches.
+  - This workflow should run on all three OS platforms - windows, ubuntu, macos. One can make use of [matrix strategy](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix) for the same.
+  - Workflow should have a build step for syntax validations and followed by a step to run the test.
+  - It is also recommended to add the test coverage tooling with your infra. The test coverage should provide details for newly added code like - code coverage percentage, file wise code coverage details.
+
+- Pull request reviewer must check for following:
+  - Code reviewers should be able to see code coverage data before approving the PR. PR approval must be done only when code coverage is more than 80%. To start with, code coverage should be measured at file level, which is each file has 80% coverage.
+
+Developer is expected to maintain the test suite copy with each branch to maintain the product code to test code compatibility.
+
+## Releasing Action:
+
+Functional tests are essential to catch bugs in preflight development branches before being released into production.
+
+Release process consists of below steps:
+
+- forking out a branch releases/Vn from the master branch
+- check-in the required dependencies
+- testing
+- tagging the commit with appropriate tag
+
+On every code push into master and releases branch always run the right set of function tests to validate the code quality. Unless all functional tests are passed, don&#39;t proceed for releasing a new version / hotfix.
+
+Running functional tests automatically on each code push can be achieved by using an action workflow.
+
+- Set workflow trigger to run on code push for master and all releases branches
+- Workflow should contain jobs per branch.
+- Each job should test action for different customer scenarios.
+- For master branch build the code and test the build action ( As master branch does not contain the supporting modules checked-in)
+- Jobs should be executed conditionally depending on the branch where code got pushed.
+- The secrets used in this workflow should be managed using [secret variables](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets). PR updating this workflow should be carefully reviewed for any malicious updates on secrets.
```

Comment:
Should have this in bold? And in PR, like we publish code coverage results, can we post a checklist for review with this point covered?

Reply:
good one, updating